### PR TITLE
Clean up autophase repo

### PR DESCRIPTION
**Pull request by hqjenny**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,36 @@
-# AutoPhase
+# AutoPhase: Compiler Phase-Ordering for HLS with Deep Reinforcement Learning
 
-## Overview 
-AutoPhase is a framework that takes a program and ﬁnds a sequence of passes that optimize the performance of the generated circuit. 
-AutoPhase is a framework that uses deep reinforcement learning (RL) to optimize the order of the passes that increase the performance of the RTL by minimizing the clock cycle count. The framework uses LLVM at the frontend and LegUp backend tools. More details are available on the paper. 
+AutoPhase is a framework that uses deep reinforcement learning (RL) to optimize the order of the passes that increase the performance of the RTL by minimizing the clock cycle count. For more details, please see:
```
Comment (AmeerHajAli) [inline README.md:3]:
```suggestion
AutoPhase is a framework that uses deep reinforcement learning (RL) to optimize the order of the passes that increase the performance of the programs by minimizing the clock cycle count. For more details, please see: 
```

File: README.md [hunk 2]
```diff
@@ -1,24 +1,36 @@
-# AutoPhase
+# AutoPhase: Compiler Phase-Ordering for HLS with Deep Reinforcement Learning
 
-## Overview 
-AutoPhase is a framework that takes a program and ﬁnds a sequence of passes that optimize the performance of the generated circuit. 
-AutoPhase is a framework that uses deep reinforcement learning (RL) to optimize the order of the passes that increase the performance of the RTL by minimizing the clock cycle count. The framework uses LLVM at the frontend and LegUp backend tools. More details are available on the paper. 
+AutoPhase is a framework that uses deep reinforcement learning (RL) to optimize the order of the passes that increase the performance of the RTL by minimizing the clock cycle count. For more details, please see: 
+- [MLSys'19 AutoPhase Paper](https://arxiv.org/pdf/2003.00671.pdf)
```
Comment (AmeerHajAli) [inline README.md:4]:
```suggestion
- [MLSys'20 AutoPhase Paper](https://arxiv.org/pdf/2003.00671.pdf)
```

File: README.md [hunk 3]
```diff
@@ -1,24 +1,36 @@
...
+- [MLSys'19 AutoPhase Presentation](https://people.eecs.berkeley.edu/~qijing.huang/2020MLSys/2020SysML_AutoPhase_Presentation.pdf)
+- [FCCM'19 AutoPhase Paper](https://arxiv.org/pdf/1901.04615.pdf)
+
+This framework takes the input of a program compiles into LLVM IR. The neural network agent takes as input the features of the program (using IR Feature Extractor), clock cycle count (using Clock-cycle Profiler), and histogram of previously applied passes. Then it outputs the prediction of the best optimization passes to apply, which is used to generate new LLVM IR.
```
Comment (AmeerHajAli) [inline README.md:8]: This is too low level details. But if you think it should be here then:
```suggestion
AutoPhase takes a program as an input and compiles into LLVM IR. The neural network agent takes as input the features of the program (using IR Feature Extractor), clock cycle count (using Clock-cycle Profiler), and histogram of previously applied passes. Then it outputs the prediction of the best next optimization pass to apply, which is used to generate a new LLVM IR.  
```

File: README.md [hunk 4]
```diff
@@ -1,24 +1,36 @@
...
   - LLVM (comes with LegUp)
   - Clang (comes with LegUp)
 
-This framework takes the input of a program compiles into LLVM IR. The neural network agent takes as input the features of the program(using IR Feature Extractor), clock cycle count(using Clock-cycle Profiler), and histogram of previously applied passes. Then it outputs the prediction of the best optimization passes to apply, which is used to generate new LLVM IR.  
 
-## Running training on AutoPhase
+Please refer to (patch/README.md) to install the LLVM patches. 
+
+We also compare RL against:
+- insertion-based greedy search / beam search (algos/greedy/greedy\_search.py)
```
Comment (AmeerHajAli) [inline README.md:23]:
```suggestion
- Insertion-based greedy search / beam search (algos/greedy/greedy\_search.py)
```

File: README.md [hunk 5]
```diff
@@ -1,24 +1,36 @@
...
-- export LEGUP_PATH=$(realpath "/path/to/legup-4.0")
-- cd ../algos/rl
-- python  ppo_ray.py
+git clone https://github.com/ucb-bar/autophase.git 
+export LEGUP_PATH=$(realpath "/path/to/legup-4.0")
+export AUTOPHASE_PATH=$(realpath "/path/to/autophase")
+cd autophase/gym-hls/
+pip install -e .
+cd ../algos/rl
+python ppo_ray.py
```
Comment (AmeerHajAli) [inline README.md:35]: did not test if this works.

**Review by AmeerHajAli — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (hqjenny) [reply README.md:35]: I tested it on my side.

**Pull request merged**